src/lib/utils/i18n.ts

```diff
@@ -0,0 +1,20 @@
+import { join } from "path"
+
+import i18nConfig from "../../../i18n.config.json"
+
+// Return a sorted list of supported locales codes, defined in `i18n.config.json`
+export const getLocalesCodes = () => i18nConfig.map((lang) => lang.code).sort()
```

Comment:
As `i18nConfig` is constant, this could be a constant as well instead of a function and calculate it just once.

```suggestion
export const localesCodes = i18nConfig.map((lang) => lang.code).sort()
```

Reply:
Makes sense, fixed in the last commit and moved to `constants.ts`